lib/ruleHelper.js

```diff
@@ -80,6 +68,156 @@ RuleHelper.prototype = {
         return allowed;
     },
 
+    /**
+     * Checks an expression in safely in an array of allowed methods
+     * @param {Object} callee function that is being called expression.tag or expression.callee
+     * @param {Array} permitted List of valid wrapping expressions
+     * @returns {boolean}
+     */
+    isAllowedCallExpression(callee, permitted) {
+        const funcName = this.getCodeName(callee);
+        let allowed = false;
+        if (funcName && permitted.indexOf(funcName) !== -1) {
+            allowed = true;
+        }
+        return allowed;
+    },
+
+    /**
+     * Captures safely any new node types that have been missed and throw when we don't support them
+     * this normalizes the passed in identifier type to return the same shape
+     */
+    normalizeMethodCall(node) {
+        let methodName;
+        let objectName;
+        switch (node.type) {
+        case "Identifier":
+            methodName = node.name;
+            break;
+        case "MemberExpression":
+            methodName = node.property.name;
+            objectName = node.object.name;
+            break;
+        case "ArrowFunctionExpression":
+            methodName = "";
+            break;
+        default:
+            this.reportUnsupported(node, "Unexpected normalizeMethodName", `unexpected ${node.type} in normalizeMethodName`);
+        }
+        return {
+            objectName,
+            methodName
+        };
+    },
+
+    /**
+     * Returns functionName or objectName.methodName of an expression
+     * @param {Object} node
+     * @returns {String} nice name to expression call
+     */
+    getCodeName(node) {
+        const normalizedMethodCall = this.normalizeMethodCall(node);
+        let codeName = normalizedMethodCall.methodName;
+        if ("objectName" in normalizedMethodCall) {
+            codeName = `${normalizedMethodCall.objectName}.${codeName}`;
+        }
+        return codeName;
+    },
+
+    /**
+     * Checks to see if a method or function should be called
+     * If objectFilters isn't present or blank array the code should not be checked
+     * If we do have object filters and the call is a function then it should not be checked
+     * @param {Object} node call expression node
+     * @param {Object} objectFilters strings that are checked as regex to match an object name
```

Comment:
Agreed I didn't want to change too much from our initial agreed format. However as I was writing this match made more sense than filter.